dockers/Screenshotter/screenshotter.js

```diff
@@ -128,17 +130,53 @@ if (seleniumURL) {
     console.log("Selenium driver in local session");
 }
 
-if (!katexURL) {
-    katexURL = "http://" + katexIP + ":" + opts.katexPort + "/";
-}
 var toStrip = "http://localhost:7936/"; // remove this from testcase URLs
 
+process.nextTick(startServer);
+var attempts = 0;
+
+//////////////////////////////////////////////////////////////////////
+// Start up development server
+
+var devServer = null;
+var minPort = 32768;
+var maxPort = 61000;
+
+function startServer() {
+    if (katexURL || katexPort) {
+        process.nextTick(tryConnect);
+        return;
+    }
+    var onerror = function() {
+        if (++attempts > 50) {
+            throw new Error("Failed to start up dev server");
+        }
+        process.nextTick(startServer);
+    };
+    var port = Math.floor(Math.random() * (maxPort - minPort)) + minPort;
+    var server = http.createServer(app).listen(port);
+    server.once("error", onerror);
+    server.once("listening", function() {
+        server.removeListener("error", onerror);
```

Comment:
I had somehow assumed that the `app` would take care of error handling from that point onward, but that's probably an incorrect assumption. On the other hand, once the server is started, we don't want any more startup attempts, so I'll have to add that distinction to the error handling function.